ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/drawing-and-display.tsx

```diff
@@ -61,6 +61,123 @@ export const removeOldDrawing = ({ map, id }: { map: any; id: string }) => {
   relevantPrimitives.length > 0 && map.getMap().scene.requestRender()
 }
 
+const makeOldDrawingNonEditable = ({ map, id }: { map: any; id: string }) => {
+  const relevantPrimitives = map
+    .getMap()
+    .scene.primitives._primitives.filter((prim: any) => {
+      return prim.id === id
+    })
+  relevantPrimitives.forEach((relevantPrimitive: any) => {
+    if (typeof relevantPrimitive.setEditMode === 'function') {
+      relevantPrimitive.setEditMode(false)
+    }
+    if (typeof relevantPrimitive.setEditable === 'function') {
+      relevantPrimitive.setEditable(false)
+    }
+  })
+  relevantPrimitives.length > 0 && map.getMap().scene.requestRender()
+}
+
+const nestedArraysOverlap = (arrayA: any[], arrayB: any[]) => {
+  let result = false
+  arrayA.map((elemA) => {
+    result =
+      result ||
+      arrayB.some((elemB) => JSON.stringify(elemA) === JSON.stringify(elemB))
+  })
+  return result
+}
+
+const isNewShape = (model: any) => {
+  const mode = model.get('mode')
+  switch (mode) {
+    case 'bbox':
+      const box = {
+        north: model.get('north'),
+        east: model.get('east'),
+        west: model.get('west'),
+        south: model.get('south'),
+      }
+      let prevModel = model.previousAttributes()
+      if (box.north && prevModel) {
+        const prevBox = {
+          north: prevModel['north'],
+          east: prevModel['east'],
+          west: prevModel['west'],
+          south: prevModel['south'],
+        }
+        if (prevBox.north) {
+          return !(
+            box.north === prevBox.north ||
+            box.east === prevBox.east ||
+            box.west === prevBox.west ||
+            box.south === prevBox.south
+          )
```

Comment:
Should these be ANDed together instead of ORed? Maybe I'm not understanding what exactly this logic should be doing, but seems like logic for determining if the bbox is different should be any coordinate not equal, so `north !== north || south !== south ||... `, and the equivalent logic for would be `!(north === north && south === south &&...)`

Reply:
Yep, your edit is correct. The negation is just so the value matches the name of the function.